File: 0075 sort-colors/solution.go
func sortColors(nums []int)  {
    low,mid,high:=0,0,len(nums)-1
    for mid<=high {
        if nums[mid]==0 {
            nums[mid],nums[low] = nums[low],nums[mid]
            low++
            mid++
        } else if nums[mid] == 1 {
            mid++
        } else {
            nums[mid],nums[high] = nums[high],nums[mid]
            high--
        }
    }
}

File: 0075 sort-colors/solution.cs
public class Solution {
    public void SortColors(int[] nums) {
        int low=0,mid=0,high=nums.Length-1;
        while(mid<=high) {
            if(nums[mid]==0) {
                int temp = nums[low];
                nums[low++]=nums[mid];
                nums[mid++]=temp;
            } else if(nums[mid]==1) mid++;
            else {
                int temp = nums[high];
                nums[high--]=nums[mid];
                nums[mid]=temp;
            }
        }
    }
}
